internal/terraform/context_plan2_test.go

```diff
@@ -2530,6 +2737,33 @@ output "a" {
 			t.Fatalf("wrong error:\ngot:  %s\nwant: %q", got, want)
 		}
 	})
+
+	t.Run("condition fail refresh-only", func(t *testing.T) {
+		plan, diags := ctx.Plan(m, states.NewState(), &PlanOpts{
+			Mode: plans.RefreshOnlyMode,
+			SetVariables: InputValues{
+				"boop": &InputValue{
+					Value:      cty.StringVal("nope"),
+					SourceType: ValueFromCLIArg,
+				},
+			},
+		})
+		assertNoErrors(t, diags)
+		if got, want := diags.ErrWithWarnings().Error(), "Module output value precondition failed: Wrong boop."; got != want {
```

Comment:
This probably isn't a big deal, but note that `diags.ErrWithWarnings()` will return `nil` if `diags` is empty, which I think would make the subsequent `.Error()` call panic here rather than emitting a test failure. I don't know that I care enough about this to warrant adding too much extra test complexity to deal with it, but just wanted to note it.

Reply:
Thanks! I didn't think of that, and I've added a few guard `Fatalf`s to catch it.